src/os/file/mime/text.js

```diff
@@ -65,22 +65,12 @@ os.file.mime.text.getText = function(buffer, opt_file) {
     var s = os.file.mime.text.hasDeclaredEncoding_(buffer, opt_file);
 
     if (!s) {
-      // I'm aware this looks dumb and doubles the memory, but jschardet lacks the ability
-      // to take a typed array as input.
-      var binaryString = '';
       var arr = new Uint8Array(buffer);
-      // Don't send jschardet more than 16KB of data. It is pretty sluggish on large files
-      var max = Math.min(arr.length, 16 * 1024);
-      for (var i = 0; i < max; i++) {
-        binaryString += String.fromCharCode(arr[i]);
-      }
-
-      var encoding = jschardet.detect(binaryString).encoding;
+      var encoding = chardetng.detect(buffer);
       if (encoding) {
         // strip any UTF BOMs before decoding
         var boms = os.file.mime.text.BOMS_;
-        var n;
-        for (i = 0, n = boms.length; i < n; i++) {
+        for (var i = 0, n = boms.length; i < n; i++) {
```

Comment:
is `n` uninstantiated here?

Reply:
No that's valid JS.
`var a = 5, b = 10, c = 15;`
`console.log('${a} ${b} ${c}');`
`Output: "5 10 15"`

Reply:
Ha, what an idiot; my bad, gotta stop doing these reviews before I eat.